rosidl_typesupport_cpp/resource/action__type_support.cpp.em

```diff
@@ -0,0 +1,68 @@
+// generated from rosidl_typesupport_cpp/resource/action__type_support.cpp.em
+// generated code does not contain a copyright notice
+
+@#######################################################################
+@# EmPy template for generating <action>__type_support.cpp files
+@#
+@# Context:
+@#  - spec (rosidl_parser.ActionSpecification)
+@#    Parsed specification of the .action file
+@#  - subfolder (string)
+@#    The subfolder / subnamespace of the message, usually 'action'
+@#  - type_supports (list of strings)
+@#    The name of the type support packages
+@#  - get_header_filename_from_msg_name (function)
+@#######################################################################
+@
+
+#include <action_msgs/msg/goal_status_array.hpp>
+#include <action_msgs/srv/cancel_goal.hpp>
+
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__feedback.hpp>
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__goal.hpp>
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__result.hpp>
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__struct.hpp>
+#include <@(spec.pkg_name)/msg/rosidl_generator_c__visibility_control.h>
+
+#include "rosidl_generator_c/action_type_support_struct.h"
+#include "rosidl_generator_cpp/action_type_support_decl.hpp"
+#include "rosidl_generator_cpp/message_type_support_decl.hpp"
+#include "rosidl_generator_cpp/service_type_support_decl.hpp"
+
+namespace @(spec.pkg_name)
+{
+
+namespace @(subfolder)
+{
+
+namespace rosidl_typesupport_cpp
+{
+
+static rosidl_action_type_support_t @(spec.action_name)_action_type_support_handle = {
+  NULL, NULL, NULL, NULL, NULL};
+
+}  // namespace rosidl_typesupport_cpp
+
+}  // namespace @(subfolder)
+
+}  // namespace @(spec.pkg_name)
+
+namespace rosidl_generator_cpp
+{
+
+template<>
+ROSIDL_GENERATOR_C_PUBLIC_@(spec.pkg_name)
```

Comment:
It looks like this will need a different visibility macro `ROSIDL_GENERATOR_C_PUBLIC_<package name>_ACTION`. This is needed to get the `dllexport` and `dllimport` correct when generating these files.

See changes in this diff.

https://github.com/ros2/rcl_interfaces/pull/47/files/a70935a2109267bb4ea7cf9ee17fdd50f3995d5b..6d6da5801ed3ee1438246bc5b31898d12d522ff0

Reply:
Good point. Added.